### ISSUE TITLE
MISP on PHP ver 7.4 clutters debug.log with notices "access array offset on value of type int"

### ISSUE THREAD

**Issue by chrisinmtown** [MISP/MISP#6544]
Check filter value type, string vs int, to avoid spurious notices from PHP 7.4 in the MISP app debug.log.

The debug.log in our 2.4.131 system has many (*many*) entries like this:
```
2020-11-02 20:39:16 Notice: Notice (8): Trying to access array offset on value of type int in [/var/www/MISP/app/Model/AppModel.php, line 2673]
Trace:
ErrorHandler::handleError() - APP/Lib/cakephp/lib/Cake/Error/ErrorHandler.php, line 232
AppModel::convert_filters() - APP/Model/AppModel.php, line 2673
Event::set_filter_eventid() - APP/Model/Event.php, line 2603
Attribute::buildFilterConditions() - APP/Model/Attribute.php, line 4508
Attribute::restSearch() - APP/Model/Attribute.php, line 4556
AppController::restSearch() - APP/Controller/AppController.php, line 1273
ReflectionMethod::invokeArgs() - [internal], line ??
Controller::invokeAction() - APP/Lib/cakephp/lib/Cake/Controller/Controller.php, line 499
Dispatcher::_invoke() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 193
Dispatcher::dispatch() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 167
[main] - APP/webroot/index.php, line 92
```
Here's the code that triggers the notice, line 2673 of AppModel.php:
```
                    if ($param[0] === '!') {
```
From what I can tell, in the usage that triggers the notice, the convert_filters function is invoked with an array that contains a single event ID (180268, for what that's worth).  I believe the code just needs to check if the parameter is a string vs integer, and not attempt to do a substring/array access on an integer.  That should silence this notice. 

According to this PHP release note:

https://www.php.net/manual/de/migration74.incompatible.php#migration74.incompatible.core.non-array-access

Only systems running PHP 7.4 or later will see this notice.

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       |  RedHat
| OS version (client)       | all
| PHP version               | 7.4.7
| MISP version / git hash   | 2.4.131
| Browser                   | If applicable

### Expected behavior

The debug.log doesn't fill with silly notices.

### Actual behavior

Lots of complaints about "Trying to access array offset on value of type int"

### Steps to reproduce the behavior

It appears that REST search triggers this.  Check debug.log file.

### Logs, screenshots, configuration dump, ...

See above.

**Comment by chrisinmtown** [comment]
I am not a PHP programmer. I tried this revision to line 2673:
```
                    if (is_string($param[0]) && $param[0] === '!') {
```
but with no success, the complaint notice still appears, so there's something I don't understand going on here.

**Comment by mokaddem** [comment]
@chrisinmtown Thanks for the report, it should be fixed on 2.4. Let us know if it's not the case!

**Comment by chrisinmtown** [comment]
Thanks for the repair @mokaddem
